bonito/decode.py

```diff
@@ -8,28 +8,54 @@
 from fast_ctc_decode import beam_search
 
 
+def phred(prob, scale=2.0, bias=0.7, offset=32):
+    """
+    Converts `prob` into a phred quality score between 1 and 40.
+
+    :returns: chr(qscore + offset)
+    """
+    if prob > 0.9999: q = 40
+    elif prob < 0.20: q = 1
+    else: q = -10 * np.log10(1 - prob) * scale + bias
+    return chr(int(max(min(q, 40), 1)) + offset)
+
+
 def decode_ref(encoded, labels):
     """
     Convert a integer encoded reference into a string and remove blanks
     """
     return ''.join(labels[e] for e in encoded if e)
 
 
-def greedy_ctc_decode(predictions, labels):
+def greedy_ctc_decode(predictions, labels, qscores=False):
```

Comment:
This implementation is now ~15X slower with `qscores=False` and ~30X when `qscores=True` which makes it slower than the model inference. We just [merged a greedy decode](https://github.com/nanoporetech/fast-ctc-decode/pull/13) into `fast-ctc-decode` so we should be able to recover speed by implementing this in rust.

Reply:
The rust implementation is over 3 orders of magnitude faster so I'll probably revert this change.

https://github.com/nanoporetech/fast-ctc-decode/pull/18